src/Piral.Blazor.Tools/tasks/SetMonorepoPiralInstanceTask.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Converters;
+using System;
+using System.IO;
+using System.Dynamic;
+using Piral.Blazor.Tools.Models;
+
+namespace Piral.Blazor.Tools.Tasks
+{
+    public class SetMonorepoPiralInstanceTask : Task
+    {
+        [Required]
+        public string PiralInstancePath { get; set; }
+
+        [Required]
+        public string PiletPath { get; set; }
+
+        public override bool Execute()
+        {
+            // System.Diagnostics.Debugger.Launch(); 
+            try
+            {
+                var piletPackageJsonFile = $"{PiletPath}/package.json";
+
+                if (!File.Exists(piletPackageJsonFile))
+                {
+                    Log.LogError($"Could not find pilet package.json file here:'{piletPackageJsonFile}'."); 
+                    return false;
+                }
+
+                var piralInstanceVersion = GetPiralInstanceVersion();
+                var piralInstancePathInPackageJson = $"file:..\\\\{PiralInstancePath.Replace("\\", "\\\\").Replace("/", "\\\\")}";
+
+                var piletPackageJsonText = File.ReadAllText(piletPackageJsonFile); 
+                piletPackageJsonText = piletPackageJsonText.Replace(piralInstancePathInPackageJson, piralInstanceVersion);
+                File.WriteAllText(piletPackageJsonFile, piletPackageJsonText); 
+            }
+            catch (Exception error)
+            {
+                Log.LogError(error.Message);  
+                return false;
+            }
+            return true; 
+        }
+
+        private dynamic GetPiralInstanceVersion(){
+            var emulatorDirectory = Path.GetDirectoryName(PiralInstancePath);
+            var piralInstanceDirectory = emulatorDirectory.Replace("\\dist\\emulator", "");
```

Comment:
Hm not sure about this part.

In the monorepo case we don't need / want to have an emulator. We can leave it for now, but it would feel awkward to generate a tgz / reference this directory when its not needed.